File: lab4/src/warehouse/resources/Fuel.java
package warehouse.resources;

import enums.FuelType;
import enums.SupplyType;
import request.SupplyRequestDetail;

public class Fuel extends Resource {
    FuelType type;

    public Fuel(int id, String name, FuelType type, int count)
    {
        super(id, name, count);
        this.type = type;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.FUEL &&
               detail.getFuelType() == type;
    }

    public FuelType getType() {
        return type;
    }
}

File: lab5/C++/src/Fuel.cpp
#include "Fuel.hpp"
#include "SupplyRequestDetail.hpp"

Fuel::Fuel(const Fuel& other) :
    Resource(rand(), other.GetName(), MeasureUnit::eLiter, other.GetQuantity()), m_type(other.m_type) { }

Fuel::Fuel(int id, const std::string& name, FuelType type, float quantity)
    : Resource(id, name, MeasureUnit::eLiter, quantity), m_type(type) {
}

bool Fuel::IsMatches(const SupplyRequestDetail& detail) const {
    return detail.GetSupplyType() == SupplyType::Fuel && detail.GetFuelType() == m_type;
}

FuelType Fuel::GetType() const {
    return m_type;
}